packages/world/src/Vec3.sol

```diff
@@ -86,17 +86,17 @@ library Vec3Lib {
 
   function x(Vec3 a) internal pure returns (int32) {
     // Extract z component (leftmost 32 bits)
-    return int32(uint32((Vec3.unwrap(a) >> 64) & 0xFFFFFFFF));
```

Comment:
why was this removed? or added in the first place if its not needed?

Reply:
yeah the mask was not needed, as we are already casting it